### ISSUE TITLE
`no-unnecessary-collection` incorrectly reports array literals containing a spread element

### ISSUE THREAD

**Issue by JasonWeinzierl** [JasonWeinzierl/eslint-plugin-rxjs-x#480]
The `rxjs-x/no-unnecessary-collection` rule incorrectly treats an array literal containing a spread element as a single-valued array.

For example:

```ts
import { combineLatest, Observable } from 'rxjs';

function combineObservables(observables: Observable<unknown>[]) {
    return combineLatest([...observables]);
}
```

`[...observables]` is not necessarily single-valued; it contains the runtime contents of `observables`, which can have zero, one, or many elements.

### Expected behavior

The rule should not report this expression as a single-valued array.

### Actual behavior

The rule reports the expression because the array literal has one AST element, even though that element is a `SpreadElement`.

### Cause

The rule appears to determine whether an array is single-valued by counting its non-null AST elements:

```ts
const nonNullElements = firstArg.elements.filter(
    (element) => element !== null,
);

if (
    nonNullElements.length === 1 &&
    couldBeObservable(nonNullElements[0])
) {
    // Report single-valued array
}
```

For `[...observables]`, `firstArg.elements.length` is `1`, but that element can expand to any number of runtime values.

### Suggested fix

Exclude spread elements when detecting statically single-valued arrays, or otherwise avoid reporting when the number of runtime elements cannot be determined:

```ts
if (
    nonNullElements.length === 1 &&
    nonNullElements[0].type !== AST_NODE_TYPES.SpreadElement &&
    couldBeObservable(nonNullElements[0])
) {
    // Report single-valued array
}
```

A regression test could cover:

```ts
// Valid
combineLatest([...observables]);

// Invalid
combineLatest([observable]);
```

Although the example can independently be simplified to `combineLatest(observables)`, the current diagnostic is still inaccurate: the spread array is not statically single-valued, and “use the observable directly” does not describe the suggested transformation.

### Environment

- `eslint-plugin-rxjs-x`: `0.9.5`
- TypeScript: `5.9`
- Node.js: `22`
